controllers/library.py

```diff
@@ -0,0 +1,164 @@
+import logging
+from models.library_items.items import LibraryItem
+from models.patrons.patron import Patron
+from pydantic import BaseModel, ValidationError
+from typing import Dict, List
+from dal.dal import insert_document_dal, delete_document_dal
+from utils.utils import patron_pydantic_to_mongoengine, item_pydantic_to_mongoengine
+from mongoengine import DoesNotExist, NotUniqueError, OperationError
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+"""
+Represents a library system with various attributes for managing its system.
+
+Attributes:
+    library_items (Dict[str, LibraryItem]): Dictionary of library items.
+    patrons (Dict[str, Patron]): Dictionary of library patrons.
+    bills (Dict[str, float]): Dictionary of patron bills.
+    name (str): The name of the library.
+"""
+
+
+class Library(BaseModel):
+    library_items: Dict[str, LibraryItem] = {}
+    patrons: Dict[str, Patron] = {}
+    bills: Dict[str, float] = {}  # Dictionary to keep track of all bills, keyed by patron ID
+    name: str
+
+    def add_new_library_items_to_the_library(self, new_library_items: List[LibraryItem]):
+        """
+        Add new library items to the library system.
+
+        Args:
+            new_library_items (List[LibraryItem]): List of library items to add.
+        """
+        try:
+            for new_library_item in new_library_items:
+                if new_library_item.isbn in self.library_items.keys():
+                    raise NotUniqueError(
+                        f"The library item with ISBN {new_library_item.isbn} already exists in the library"
+                    )
+                self.library_items[new_library_item.isbn] = new_library_item
+                new_library_item_document = item_pydantic_to_mongoengine(new_library_item)
+                success, error_message = insert_document_dal('library-items', new_library_item_document)
+                if not success:
+                    raise ValueError(error_message)
+                logging.info(
+                    f"The library item {new_library_item.title} with ISBN {new_library_item.isbn} was added to '{self.name}' library"
+                )
+        except NotUniqueError as e:
+            logging.error(f"Add library item failed: {e}")
+            raise ValueError(e)
+        except ValidationError as e:
+            logging.error(f"Validation error: {e}")
+            raise ValueError(f"Validation error: {e}")
```

Comment:
why value error create LirbraryError with original error and message as attributes